src/Altinn.Broker.API/Helpers/IdempotencyEventHelper.cs

```
@@ -1,29 +1,32 @@
 using Altinn.Broker.Application;
+using Altinn.Broker.Application.ConfirmDownloadCommand;
 using Altinn.Broker.Core.Domain;
 using Altinn.Broker.Core.Repositories;
 
 using OneOf;
 
 namespace Altinn.Broker.Helpers;
 
-public class WebhookEventHelper
+public class IdempotencyEventHelper
 {
-    public static async Task<OneOf<Task, Error>> ProcessMalwareEvent(ScanResultData data, MalwareScanningResultHandler handler, IWebhookEventRepository webhookEventRepository, CancellationToken cancellationToken)
+    public static async Task<OneOf<Task, Error>> ProcessEvent(string uniqueString, Func<Task<OneOf<Task, Error>>> process, IIdempotencyEventRepository idempotencyEventRepository, CancellationToken cancellationToken)
     {
         try
         {
 
             // Create a new entry for that webhook id
-            await webhookEventRepository.AddWebhookEventAsync(data.ETag, cancellationToken);
+            Console.WriteLine("Adding idempotency event");
+            await idempotencyEventRepository.AddIdempotencyEventAsync(uniqueString, cancellationToken);
             try
             {
-                // Call you method
-                return await handler.Process(data, cancellationToken);
+                // Call you method¨
+                Console.WriteLine("Processing event");
+                return await process();
             }
             catch (Exception e)
```

Should all exceptions be swallowed here or should they be re-thrown?